### ISSUE TITLE
How to convert existing VM into template from vmware driver

### ISSUE THREAD

**Issue by anandnevase** [saltstack/salt#32855]
### Description of Issue/Question

How to convert existing VM into template from vmware driver?? Similar thing we can do from vCenter GUI.

Currently we can clone a VM and create template out of it using following config:

```
$ cat /etc/salt/cloud.profiles 
vm_profile:
  provider: vcenter
  clonefrom: test-vm
  template: true
  folder: 'VMs'
  deploy: False

$ salt-cloud -p  vm_profile  'vm_template' -l debug
```

But how can i convert already existing VM to Template??

**Comment by jfindlay** [comment]
Ping @nmadhok.  @anandnevase, you may also want to ask on the [salt-users](https://groups.google.com/forum/#!forum/salt-users) mailing list.

**Comment by nmadhok** [comment]
@anandnevase @jfindlay That functionality currently doesn't exist but should be pretty easy to add (since the code for it already exists). You can currently clone a VM to a template and then destroy the original VM.

**Comment by anandnevase** [comment]
@nmadhok Creating template using VM cloning is not good option. As per [pyvmomi documentation](https://github.com/vmware/pyvmomi/blob/master/docs/vim/VirtualMachine.rst), we can use ```MarkAsTemplate()``` method to convert existing VM into template.

**Comment by nmadhok** [comment]
@anandnevase Will look into adding the functionality sometime soon.

**Comment by fishhead108** [comment]
Fixed in #43980, could be close now.
